src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,187 @@
+
+                    // Assert that it contains all the query parameters with appropriate type
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$select") && param.Schema.Type.Equals("string")));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$filter") && param.Schema.Type.Equals("string")));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$orderby") && param.Schema.Type.Equals("string")));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$first") && param.Schema.Type.Equals("integer")));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$after") && param.Schema.Type.Equals("string")));
+                }
+                else
+                {
+                    // No query parameters for other OperationTypes. 
+                    Assert.IsFalse(operation.Parameters.Any(param => param.In is ParameterLocation.Query));
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Validates that Input parameters are generated for Stored Procedures with GET operation.
+    /// Input parameters for stored procedures will be generated for GET operation.
+    /// If a parameter default value is not provided in the config, it will marked as required.
+    /// It also validates parameter metadatas like type, name, location, required, etc.
+    /// </summary>
+    /// <param name="entityName">The name of the entity.</param>
+    /// <param name="objectName">The name of the database object.</param>
+    [DataTestMethod]
+    [DataRow("CountBooks", "count_books", DisplayName = "StoredProcudure with no parameters results in 0 created input query params.")]
+    [DataRow("UpdateBookTitle", "update_book_title", DisplayName = "StoredProcedure with parameters results in created input query params.")]
+    public async Task ValidateInputParametersForStoredProcedures(string entityName, string objectName)
+    {
+        Dictionary<string, object> parameterDefaults = null;
+        if (entityName.Equals("UpdateBookTitle"))
+        {
+            // Adding a parameter default value.
+            parameterDefaults = new Dictionary<string, object> { { "title", "Test Title" } };
+        }
+
+        EntitySource entitySource = new(Object: objectName, EntitySourceType.StoredProcedure, parameterDefaults, null);
+        OpenApiDocument openApiDocument = await GenerateOpenApiDocumentForGivenEntityAsync(entityName, entitySource);
+
+        OpenApiPathItem pathItem = openApiDocument.Paths.First().Value;
+        foreach ((OperationType operationType, OpenApiOperation operation) in pathItem.Operations)
```

Comment:
You'll probably want to validate that `pathItem.Operations` contains the expected element and expected number of elements. (This is assuming that test will pass when zero assertions are hit.) This would indicate a 'false-negative'. In other words the test would pass (negative for any issue detected) and we would have false confidence that the expected behavior occurred during open api doc generation. 

I realize this comment applies to other pre-existing tests outside the scope of this pr, and we'll have to go fix those.